Edited (#412)
vaadin/flow-and-components-documentation: documentation/components/tutorial-flow-components-setup.asciidoc

= Using Vaadin Components

> The Vaadin Platform includes a set of prebuilt server-side https://vaadin.com/components/browse[components], with Java APIs, that you can use to build your UI.   

Reviewer: ```suggestion
The Vaadin Platform includes a set of https://vaadin.com/components/browse[components], with server-side Java APIs, that you can use to build your UI.   
```

The components are included, together with Flow, as dependencies in platform.

> Here is a list of available components:

Reviewer: ```suggestion
Here is a list of components included in _vaadin-core_ module, which includes all open source components:
```

- `Upload` - https://vaadin.com/components/vaadin-upload/java-examples[Demos], <<tutorial-flow-upload#,Documentation>>, https://vaadin.com/api/platform/com/vaadin/flow/component/upload/Upload.html[Javadoc], https://github.com/vaadin/vaadin-upload-flow[sources and issue tracking]

== Vaadin Platform Dependency

The components are part of the Vaadin platform and are included as dependencies, together with Flow. Each component has a Java API. 

> Using the platform dependency ensures that all available components are included automatically. You are guaranteed to get compatible versions of both Flow and the components.

Reviewer: ```suggestion
Using the platform dependency (com.vaadin:vaadin) ensures that all available components, both open source and commercial, are included automatically. You are guaranteed to get compatible versions of both Flow and the components.
```

    </dependency>
</dependencies>
----

[NOTE]
It is possible to include components individually by declaring a dependency for each component. See below for an example.

The platform also includes theming for the components that you can edit to ensure the styling suits your UI. See <<../theme/using-component-themes#,Using Component Themes>> for more.

> You can find additional prebuilt Web Components (with Java APIs) contributed by the Vaadin community in the https://vaadin.com/directory/search?framework=Vaadin%2010[Vaadin Directory.]

Reviewer: @samie What should be the link here? Should we use Vaadin 10 or something else as a filter🤔

Reviewer: Removed the search filter

> You can find additional prebuilt Web Components (with Java APIs) contributed by the Vaadin community in the https://vaadin.com/directory/search?framework=Vaadin%2010[Vaadin Directory.]

Reviewer: Let's just leave out the search parameter altogether at this point as Directory don't support anything sane that keeps up to date ATM.
```suggestion
You can find additional prebuilt Web Components (with Java APIs) contributed by the Vaadin community in the https://vaadin.com/directory/search[Vaadin Directory.]
```

</dependencies>
----

== Using WebJars

> Individual component packages consists of two main sections:

Reviewer: ```suggestion
Individual component packages often consists of two parts:
```

. Java API classes.
. Web Component files (HTML, JavaScript, CSS ..).

> Both sections are provided in the same JAR by default, and both sections are used to display the components correctly in pages.

Reviewer: This part needs some rewriting. java api and the web component are distrubuted in different jar files. And both are needed. vaadin-foobar-flow depends on vaadin-foobar, so developers will automatically get the vaadin-foobar to their project, just by defining they want vaadin-foobar-flow to their application. 

One option would be to remove this line altogether.

> Both sections are provided in the same JAR by default, and both sections are used to display the components correctly in pages.

Reviewer: ```suggestion
The Java API modules have a dependency to the Web Component files, so Flow users don't typically need to worry about them at all.
```

> Web Component files are provided as https://github.com/webjars/webjars/[WebJars] (JAR files that contain Web Component files), the design of which was heavily influenced by the Polymer package manager, https://bower.io/[Bower] (not developed or maintained by Vaadin). WebJars facilitate using Maven, instead of Bower, for the same purpose. 

Reviewer: Wow, this was rather oddly put (already before the edits). I'll try to rewrite it.

```suggestion
Web Component files are provided as https://github.com/webjars/webjars/[WebJars] (JAR files that contain Web Component files). WebJars contain the resources needed by the client-side and it also contain the dependency information as Maven pom.xml files, so there is no need to use Bower or similar for the client side dependency management. 
```

> Flow is able to resolve requests into WebJar-compatible content by imitating regular Web Component files.

Reviewer: I'd just remove this line altogether. Flow is not imitating anything and there are no "regular Web Component files", but Flow just serves the client side requests from the archive when asked.

```suggestion
```

Using WebJars does not restrict you from also using external Web Component files, and this is enabled by default. If no suitable WebJar is found, request resolving falls back to the configured external Web Component location. You can turn off WebJar resolving by changing your configuration. 

> [NOTE]

Reviewer: ```suggestion
```

> Current https://github.com/webjars/webjars/issues[WebJar limitations] require extra maven configurations: extra repository + `bom` file declaration.

Reviewer: I have no idea what this means. Unless @pleku tells my what this means, let's just remove this note.

Reviewer: ```suggestion
```

> === Excluding WebJars

Reviewer: @pleku Why is this here in the first place? When would you like to do this and why is it so special that it needs to be mentioned in the docs?
